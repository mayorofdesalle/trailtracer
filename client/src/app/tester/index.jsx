import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
