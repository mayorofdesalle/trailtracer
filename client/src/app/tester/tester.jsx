import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };