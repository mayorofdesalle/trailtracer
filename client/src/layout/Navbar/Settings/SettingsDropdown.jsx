import PropTypes from 'prop-types';
import { forwardRef } from 'react';
import styled from 'styled-components';

import Dropdown from '@components/ui/Dropdown';

import LanguageSelector from './LanguageSelector';
import ThemeToggle from './ThemeToggle';

const SettingsDropdownInner = styled(Dropdown)`
    height: clamp(7.5rem, min(16dvw, 16dvh), 29rem);
    width: clamp(10.5rem, min(22dvw, 22dvh), 41rem);
    top: clamp(2.25rem, min(5dvw, 5dvh), 5.5rem);
    right: 0;
    padding: clamp(1rem, min(2dvw, 2dvh), 4rem) clamp(1rem, min(2dvw, 2dvh), 4rem) clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    justify-content: space-around;
    overflow: visible;
`;

/**
 * SettingsDropdown
 * @param {function} close - The function to close the dropdown.
 * @param {object} props - The rest of the native props of the component.
 * @description
 * Dropdown to display the settings.
 **/
const SettingsDropdown = forwardRef(function SettingsDropdown({ close, ...props }, ref) {

    return (
        <SettingsDropdownInner ref={ref} onBlur={close} {...props}>
            <ThemeToggle />
            <LanguageSelector />
        </SettingsDropdownInner>
    );
});

SettingsDropdown.propTypes = {
    close: PropTypes.func.isRequired
};

export default SettingsDropdown;