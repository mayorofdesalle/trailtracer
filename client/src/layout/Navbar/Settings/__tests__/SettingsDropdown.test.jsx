import { describe, expect, it, render, screen, vi } from '@app/tester';

import SettingsDropdown from '../SettingsDropdown';

describe('SettingsDropdown', () => {

    it('should render correctly', () => {
        render(<SettingsDropdown close={vi.fn()} data-testid='settings-dropdown' />);
        const dropdown = screen.getByTestId('settings-dropdown');
        expect(dropdown).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<SettingsDropdown close={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

});