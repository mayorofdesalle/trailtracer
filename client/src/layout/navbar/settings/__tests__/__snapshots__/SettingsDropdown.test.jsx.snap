// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsDropdown > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-hLQSwg sc-fmKFGs kHwwsn fkKYEp lgqhfG jmMOyH"
    tabindex="0"
  >
    <div
      class="sc-guDLey sc-ktwOfi kHwwsn foUNKg"
    >
      <div
        class="sc-gLLuof dcSHkq"
      >
        forms.theme
      </div>
      <div
        class="sc-guDLey sc-hABBmJ kHwwsn fOpcPM"
      >
        <button
          aria-checked="false"
          class="sc-beySPh zbkUW"
          role="switch"
          type="button"
        >
          <svg
            class="sc-eDLKkx kBnHHP"
            fill="#435049"
            role="img"
          >
            <title>
              sun-fill
            </title>
            <use
              href="http://localhost:3000/src/assets/icons.svg#sun-fill"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="sc-guDLey sc-brSamD kHwwsn euRBwi"
    >
      <div
        class="sc-gLLuof dcSHkq"
      >
        forms.language
      </div>
      <div
        class="sc-csKJxZ sc-eTNRI jxMgKP hsvJbM css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="Select__control css-cp01gg-control"
        >
          <div
            class="Select__value-container css-1qpkzkj-ValueContainer"
          >
            <div
              class="Select__placeholder css-1vlsb4t-placeholder"
              id="react-select-3-placeholder"
            >
              Select...
            </div>
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-describedby="react-select-3-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              aria-readonly="true"
              class="css-mohuvp-dummyInput-DummyInput"
              id="react-select-3-input"
              inputmode="none"
              role="combobox"
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="Select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="Select__indicator-separator css-j4w2j1-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="Select__indicator Select__dropdown-indicator css-g56vrd-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
